sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/hash.scala

```diff
@@ -369,11 +369,25 @@ abstract class HashExpression[E] extends Expression {
   protected def genHashBoolean(input: String, result: String): String =
     genHashInt(s"$input ? 1 : 0", result)
 
-  protected def genHashFloat(input: String, result: String): String =
-    genHashInt(s"Float.floatToIntBits($input)", result)
+  protected def genHashFloat(input: String, result: String): String = {
+    s"""
+       |if($input == -0.0f) {
+       |  ${genHashInt(s"Float.floatToIntBits(0.0f)", result)}
```

Comment:
Although this has the semantic, shall we use simply `"0"` instead of `s"Float.floatToIntBits(0.0f)"`? We may add some comment for the semantic instead.
```
jshell> Float.floatToIntBits(0.0f)
$1 ==> 0
```

Reply:
You are right, I have tested it to be sure.
`Murmur3HashFunction.hashInt(java.lang.Float.floatToIntBits(0.0f), 42) == Murmur3HashFunction.hashInt(0, 42)`
It is simpler.